### ISSUE TITLE
UC mode detected on PixelTest (Custom Script and Test Scripts)

### ISSUE THREAD

**Issue by anthonyg45157** [seleniumbase/SeleniumBase#2251]
Hello from youtube. I've been working on a scraping bot that uses seleniumbase and I need a little assistance with. It seems when I check with pixelscan and other tools my bot is being detected. Even when using a simple script., it seems to get detected.  Here is the Frankenstein script I mentioned.  I have replaced actual site with blahblahblah.com but for reference this basic concept is this:

**Python script that will collect card information for a trading card game, extract that info to a data frame, hold it, then search google with that information. Once the google search is complete the bot navigates to a website to get price data for that card relative the condition of the card. in the end it exports all of the information to an organized CSV file.**

You will probably think this code is hilarious as I used many things I now realized are built into seleniumbase and I probably didn't need some of them such as time and maybe even beautifulsoup (still learning a ton). 

Anyways here is the script. You can modify it to stay open at the end and navigate to pixelscan if you want to test but you might know right away what could be going on .  It might be better to troubleshoot with a simple script but I thought you might find this one insteresting and would love feedback

```
import time
import random
from seleniumbase import SB
from bs4 import BeautifulSoup
import pandas as pd

# Function to extract data from a given URL
def extract_data_from_url(url, data_frame):
    with SB(uc=True, incognito=True) as sb:
        sb.get(url)
        time.sleep(random.uniform(5, 8))  # Random delay after opening the URL
        page_source = sb.get_page_source()
        soup = BeautifulSoup(page_source, 'html.parser')
        data = {}
        label_selectors = {
            "Cert #": "div.certlookup-intro dl:nth-child(2) dd",
            "Card Name": "div.certlookup-intro dl:nth-child(3) dd",
            "Year": "div.certlookup-intro div:nth-child(4) dl:nth-child(2) dd",
            "Language": "div.certlookup-intro div:nth-child(4) dl:nth-child(3) dd",
            "Card Set": "div.certlookup-intro div:nth-child(5) dl:nth-child(1) dd",
            "Card Number": "div.certlookup-intro div:nth-child(5) dl:nth-child(2) dd",
            "Grade": "div.related-info.grades dl dd"
        }
        for label, selector in label_selectors.items():
            element = soup.select_one(selector)
            data[label] = element.text.strip() if element else "Data not found"
        data_frame = data_frame._append(data, ignore_index=True)
        return data_frame

# Function to get the price for a specific grade from the page source of blahblahblah123.com
def get_price_for_grade(page_source, grade):
    soup = BeautifulSoup(page_source, 'html.parser')
    price_rows = soup.select("#full-prices table tr")
    for row in price_rows:
        cells = row.find_all('td')
        if cells:
            grade_cell = cells[0].get_text(strip=True)
            price_cell = cells[1].get_text(strip=True)
            # Check for the special case of grade 10
            if grade == '10' and 'PSA 10' in grade_cell:
                return price_cell
            elif grade_cell == f"Grade {grade}":
                return price_cell
    return "Price not found"

# Read a list of URLs from a file (e.g., "urls.txt")
with open("urls.txt", "r") as url_file:
    urls = url_file.read().splitlines()

# Create an empty DataFrame to store the data
data_frame = pd.DataFrame(columns=["Cert #", "Year", "Language", "Card Set", "Card Name", "Card Number", "Grade", "Price"])

# Iterate through the list of URLs and extract data from each
for url in urls:
    print(f"Extracting data from URL: {url}")
    data_frame = extract_data_from_url(url, data_frame)

    # Construct the Google URL based on the mapping
    google_mapping = {
        1: "Card Name",
        2: "Year",
        3: "Language",
        4: "Card Set",
        5: "Card Number"
    }

    google_url = "https://www.google.com/search?q=" + "+".join([data_frame.iloc[-1][google_mapping[i]] for i in range(1, 6)])

    # Open the Google URL and click on the first hyperlink that goes to www.blahblahblah.com
    with SB(uc=True, incognito=True) as sb:
        sb.get(google_url)
        time.sleep(random.uniform(5, 8))  # Random delay before searching
        sb.wait_for_element('a[href*="www.blahblahblah.com"]')
        first_blahblahblah_link = sb.find_element('a[href*="www.blahblahblah.com"]')
        first_blahblahblah_link.click()
        time.sleep(random.uniform(5, 8))  # Random delay after clicking
        sb.wait_for_ready_state_complete()
        blahblahblah_page_source = sb.get_page_source()

    # Get the price for the grade from the page source of blahblahblah.com
    grade = data_frame.iloc[-1]['Grade']
    # Special handling for grade 10
    grade = '10' if 'Gem Mint 10' in grade else grade
    price = get_price_for_grade(blahblahblah_page_source, grade)
    data_frame.at[data_frame.index[-1], 'Price'] = price
    time.sleep(random.uniform(5, 8))  # Random delay after processing each URL

# Export the DataFrame to a CSV file
data_frame.to_csv('collected_data.csv', index=False)

# If you want to keep the browser open for inspection, otherwise remove this line
# sb.sleep(999)

```

So I decided to try a more simple script and still cannot get it to pass detections 

```
from seleniumbase import SB

# Create a function to navigate to the specified URL and keep the browser open
def navigate_to_url_and_keep_open(url):
    with SB(uc=True, incognito=True) as sb:
        # Open the specified URL
        sb.open(url)
        sb.sleep(999)

# Specify the URL you want to navigate to
url = "www.pixelscan.net"

# Navigate to the specified URL and keep the browser open
navigate_to_url_and_keep_open(url)

```

I attempted sb.get on the above script with not luck as well.  Confirmed I am using latest seleniumbase.

Any ideas, thoughts and recommendations are very very welcomed. I'm super new at python and scripts as a whole but have been spending a ton of time digging in with my spare time. 

Appreciate your hard work and dedication!

**Comment by mdmintz** [comment]
Probably fixed in `4.21.0`, now available: https://github.com/seleniumbase/SeleniumBase/releases/tag/v4.21.0

Try with the UC Mode examples, which include CF page loads, and clicking CAPTCHAs successfully:
* [SeleniumBase/examples/verify_undetected.py](https://github.com/seleniumbase/SeleniumBase/blob/master/examples/verify_undetected.py)
* [SeleniumBase/examples/raw_turnstile.py](https://github.com/seleniumbase/SeleniumBase/blob/master/examples/raw_turnstile.py)
* [SeleniumBase/examples/raw_form_turnstile.py](https://github.com/seleniumbase/SeleniumBase/blob/master/examples/raw_form_turnstile.py)

See each of those for advanced methods. If you have to perform actions after you get onto a website (while still remaining undetected) try the `driver.reconnect(seconds)` option a lot.

For advanced stealth actions, see https://github.com/seleniumbase/SeleniumBase/discussions/2248#discussioncomment-7501491 (which talks about `selenium-stealth`: an external library for masking lots of things, which is compatible with SeleniumBase). You've got some flexibility on what you can do. And there's CDP for other advanced actions.

**Comment by anthonyg45157** [comment]
Confirmed this is fixed in 4.21.0 (the ol cat and mouse game!)

Appreciate you providing examples on this as well! Keep up the great work!
